### ISSUE TITLE
머신러닝과 딥러닝

### ISSUE THREAD

**Issue by sghong977** [sghong977/AIML_Basics#1]
1. 머신러닝과 딥러닝.
관련글
- 딥러닝: [링크](http://www.tcpschool.com/deep2018/deep2018_deeplearning_intro)
- 머신러닝: [링크](https://blog.naver.com/mhn_noonai/222285873816)
- 머신러닝: [링크](https://tensorflow.blog/%ED%95%B8%EC%A6%88%EC%98%A8-%EB%A8%B8%EC%8B%A0%EB%9F%AC%EB%8B%9D-1%EC%9E%A5-2%EC%9E%A5/1-3-%EB%A8%B8%EC%8B%A0%EB%9F%AC%EB%8B%9D-%EC%8B%9C%EC%8A%A4%ED%85%9C%EC%9D%98-%EC%A2%85%EB%A5%98/)

딥러닝은 머신러닝 방법 중 하나로, neural network을 기반으로함.
MultiLayer Perceptron같은거 생각하면 되는데, 신경망 가중치(weight)를 입력에 곱하고 bias를 더하여 output을 내는 구조가 여러 레이어 쌓여있다고 보면 된다. loss function을 정의하고 forward 후에 backpropagation하여 각 learnable parameter들에 대해 loss를 편미분해서 gradient descent하면서 loss minimization하는 방향으로 learning rate만큼 학습하는 것.
다른 유명한 머신러닝 알고리즘들에 비해서 데이터를 많이 필요로 하다보니까, 최근들어 장비 가격이나 데이터에 대한 접근성이 완화되어서야 유행한 듯.

머신러닝은 데이터에 기반으로하여 규칙성, 패턴을 찾는 알고리즘 혹은 통계적 모델이다. neural network 말고 일반적인 머신러닝 알고리즘도 다양하다. 되게 다양한게 종류를 크게 2가지로 나눈다.
supervised, unsupervised learning. reward를 줘서 학습하는 reinforcement learning도 있는데 일단 이건 논외로 한다.
supervised learning에는 linear, logistic regression이나 Naive bayes, SVM, K-nearest-neighbor, Decision Tree, Random Forest (classification)같은 알고리즘이 있다.
unsupervised learning에는 K-means clustering 등이 있다.   (내가 비전 위주의 딥러닝을 다뤘다보니 이쪽은 안써서, 조금 나중에 쓰겠다.)

앞으로 정리할때 Ian Goodfellow- Deep Learning 책 목차 보면서 떠오르는대로 정리해도 될듯.
https://github.com/janishar/mit-deep-learning-book-pdf/tree/master/chapter-wise-pdf
